test/src/x509.rs

```diff
@@ -0,0 +1,226 @@
+// Licensed under the Apache-2.0 license
+
+use asn1::{ObjectIdentifier, Utf8String};
+
+#[derive(Eq, PartialEq)]
+pub struct DiceFwid {
+    pub hash_alg: asn1::ObjectIdentifier,
+    pub digest: Vec<u8>,
+}
+impl std::fmt::Debug for DiceFwid {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("DiceFwid")
+            .field("hash_alg", &format!("{}", &self.hash_alg))
+            .field("digest", &format!("{:02x?}", self.digest))
+            .finish()
+    }
+}
+
+#[derive(Debug, Default, Eq, PartialEq)]
+pub struct DiceTcbInfo {
+    pub vendor: Option<String>,
+    pub model: Option<String>,
+    pub version: Option<String>,
+    pub svn: Option<u32>,
+    pub layer: Option<u32>,
+    pub index: Option<u32>,
+    pub fwids: Vec<DiceFwid>,
+    pub flags: Option<u32>,
+    pub vendor_info: Option<Vec<u8>>,
+    pub ty: Option<Vec<u8>>,
```

Comment:
I presume this is shortened because `type` is a keyword - could this be `type_`?

Reply:
Yes. type_ might be a better option, but I've been using ty for this in quite a few other places in the codebase (mostly in ureg and systemrdl libraries).

Reply:
Guess we can change it in a follow-up PR.